### PR TITLE
Improve stability

### PR DESCRIPTION
**Pull request by m3gg3r**

### CODE REVIEW

File: src/Worker/ShopwareConsoleCaller.php [hunk 1]
```diff
@@ -57,18 +57,19 @@ public function call($command, $parameters = [])
         ];
 
         $process = \proc_open($fullCommand, $stdDescriptors, $pipes, $this->workingDirectory, null);
+        if (false === $process || false === is_resource($process)) {
+            $this->error = printf('Could not start command "%s" correctly. No valid process resource was returned', $command);
+            return false;
+        }
+
+        stream_set_blocking($pipes[0], 0);
```
Comment (michz) [inline src/Worker/ShopwareConsoleCaller.php]: For better readability:
```suggestion
        stream_set_blocking($pipes[0], false);
```
?
The second parameter is a bool: http://php.net/manual/de/function.stream-set-blocking.php

File: src/Worker/ShopwareConsoleCaller.php [hunk 2]
```diff
@@ -151,7 +152,34 @@ public function hasError()
...
     {
         $this->output = '';
-        $this->returnCode = '';
+        $this->returnCode = -1;
         return $this;
     }
+
+    /**
+     * @param resource $process     process which should be checked for exit code
+     * @param int      $maxWaitTime max time to wait for a running process (in microseconds)
```
Comment (michz) [inline src/Worker/ShopwareConsoleCaller.php]: I'm not sure if I understand that correctly. But if you really mean microseconds, do you think that 5000 is a good default value? That would be only 5 milliseconds...

Comment (m3gg3r) [reply src/Worker/ShopwareConsoleCaller.php]: Yeah, mixed up micro and milli ... will correct this

Comment (m3gg3r) [reply src/Worker/ShopwareConsoleCaller.php]: But i think, 5 seconds are enough?

Comment (michz) [reply src/Worker/ShopwareConsoleCaller.php]: Not sure. With empty cache Shopware takes sometimes longer tahn 5 seconds...

**Pull request merged**